client/analytics/components/report-chart/index.js

```diff
@@ -27,17 +29,43 @@ import { Chart, ChartPlaceholder } from 'components';
 import { getReportChartData } from 'store/reports/utils';
 import ReportError from 'analytics/components/report-error';
 
-class ReportChart extends Component {
+export const DEFAULT_FILTER = 'all';
+
+export class ReportChart extends Component {
+	getSelectedFilter( filters, query ) {
+		if ( filters.length === 0 ) {
+			return null;
+		}
+
+		const filterConfig = filters.pop();
+
+		if ( filterConfig.showFilters( query ) ) {
+			const allFilters = flattenFilters( filterConfig.filters );
+			const value = query[ filterConfig.param ] || DEFAULT_FILTER;
+			const selectedFilter = find( allFilters, { value } );
+			const selectedFilterParam = get( selectedFilter, [ 'settings', 'param' ] );
+
+			if ( ! selectedFilterParam || Object.keys( query ).includes( selectedFilterParam ) ) {
+				return selectedFilter;
+			}
+		}
+
+		return this.getSelectedFilter( filters, query );
```

Comment:
This code will never reach the second item in the `filters` array, ie `variationsConfig`: 

https://github.com/woocommerce/wc-admin/blob/c1e5f4c19cb72242df73182b0cec7d53bbc361f5/client/analytics/report/products/config.js#L116

The `value: 'all'` filter will be selected by default.


You could simplify by importing the first item directly from config to avoid recursively working through the array

Reply:
We are starting by the end of the array (https://github.com/woocommerce/wc-admin/pull/884/files/c1e5f4c19cb72242df73182b0cec7d53bbc361f5#diff-e634ee55f27a62796c86de6d587b69b1R40). We test if that filter is visible and if it's applied (https://github.com/woocommerce/wc-admin/pull/884/files/c1e5f4c19cb72242df73182b0cec7d53bbc361f5#diff-e634ee55f27a62796c86de6d587b69b1R42), otherwise we test the previous filters in the array. So it should always find the most specific filter applied.

In the Product report, for example, it will first find if a _Variations_ filter is applied and if not, will find if a _Products_ filter is.

> You could simplify by importing the first item directly from config to avoid recursively working through the array

But that wouldn't handle the case where a subfilter has a different display mode than the parent filter, right?

Reply:
The links you supplied aren't around anymore, but I think I know what you're referring to. 

>In the Product report, for example, it will first find if a Variations filter is applied and if not, will find if a Products filter is.

Gotcha, I was thinking the Variations filter was always going to be the `standard` mode for some reason. Makes sense

Reply:
> The links you supplied aren't around anymore, but I think I know what you're referring to.

My bad, I did a `git push -f` after posting the links. :sweat_smile: